patches/unique-db.diff

```diff
@@ -26,23 +26,25 @@ Index: code-server/lib/vscode/src/vs/workbench/services/storage/browser/storageS
  
  export class BrowserStorageService extends AbstractStorageService {
  
-@@ -67,7 +68,7 @@ export class BrowserStorageService exten
+@@ -67,7 +68,11 @@ export class BrowserStorageService exten
  					return `global-${this.profileStorageProfile.id}`;
  				}
  			case StorageScope.WORKSPACE:
 -				return this.payload.id;
++				// Add a unique ID based on the current path for per-workspace databases.
++				// This prevents workspaces on different machines that share the same domain
++				// and file path from colliding (since it does not appear IndexedDB can be
++				// scoped to a path) as long as they are hosted on different paths.
 +				return this.payload.id + '-' + hash(location.pathname.toString().replace(/\/$/, "")).toString(16);
```

Comment:
Will this commit affect the migration/state collisions e2e test we have?

Reply:
Nope it operates the same way; things just moved around a little bit.